requirements.txt

```diff
@@ -1,26 +1,27 @@
-aiofiles
-aiohttp
-altair>=4.2.0
+aiofiles>=22.0,<24.0
+aiohttp~=3.0
+altair>=4.2.0,<6.0
 fastapi
 ffmpy
 gradio_client>=0.2.7
 httpx
 huggingface_hub>=0.14.0
-Jinja2
+Jinja2<4.0
 markdown-it-py[linkify]>=2.0.0
-pygments>=2.12.0
 mdit-py-plugins<=0.3.3
-markupsafe
-matplotlib
-numpy
-orjson
-pandas
-pillow
-pydantic
-python-multipart
+markupsafe~=2.0
+matplotlib~=3.0
+numpy~=1.0
+orjson~=3.0
+packaging
+pandas>=1.0,<3.0
+pillow>=8.0,<11.0
+pydantic>=1.0,<3.0
```

Comment:
Nit: Let's use the same [pin](https://github.com/tiangolo/fastapi/blob/master/pyproject.toml#L45) as fastapi. Version 1.0 is actually not allowed.

Reply:
Interesting. Do we have to do this? I don't think we need to use the same versions of fastapi (that's a bit redundant anyways since fastapi is one our dependencies). We just need to make sure that our library's use of `pydantic` is valid according to the version of `pydantic` installed on user's machines. 

That being said, I haven't tested different versions of `pydantic` so maybe we actually do need to be more strict